### PR TITLE
API 5.0 WP 1+2 (API Servers & Webhooks)

### PR DESCRIPTION
**Pull request by Bibo-Joshi**

### CODE REVIEW

File: telegram/bot.py [hunk 1]
```diff
@@ -2592,18 +2598,26 @@ def set_webhook(
             data['max_connections'] = max_connections
         if allowed_updates is not None:
```
Comment (Poolitzer) [inline telegram/bot.py:2599]: here we use `is not None`, done there you use just `if:`. We probably should agree on one and go with it everywhere

**Review by Poolitzer — COMMENTED** [review]
LGTM

File: telegram/bot.py (hunk 1 above)
Comment (Bibo-Joshi) [reply telegram/bot.py:2599]: I think the important part is, that `[]` evaluates to `False`, but passing an empty list as a different meaning than not passing the value. So, IMO it's okay to use `is not None` if we need to make a difference between `bool(var) is False` and `var is None` (e.g. for booleans) and just use `if` in the other cases …

Comment (Poolitzer) [reply telegram/bot.py:2599]: but max_connections is an int?

Comment (Bibo-Joshi) [reply telegram/bot.py:2599]: and `bool(0) is False`, while `0 is not None`, so that fits my argumentation :D

Comment (Bibo-Joshi) [reply telegram/bot.py:2599]: but you're right that we're not very consistent, e.g. in `send_message`, there's a

```python
        if disable_web_page_preview:
            data['disable_web_page_preview'] = disable_web_page_preview
```

is it really that bad, though? 🤔

Comment (Poolitzer) [reply telegram/bot.py:2599]: I personally do like consistency. But I dont think it really matters, no. Just saw that and commented.

Comment (Bibo-Joshi) [reply telegram/bot.py:2599]: that's the right thing to do :) I guess it's not problematic mast times, as `None` and `False` usually mean the same to TG. but I'll try to use `if … is not None` wher it makes sence, as that's safer.

Comment (Poolitzer) [reply telegram/bot.py:2599]: alright

**Pull request merged**